.github/workflows/test.yaml

```diff
@@ -122,7 +134,7 @@ jobs:
         id: install_no_lock
         run: |
           mkdir -p .ci-package-locks/code-quality
-          pip install ".[dev]" "click<8.1.4" "matplotlib<3.8.0"
+          pip install pre-commit
```

Comment:
Should this change be made in a separate PR or at least commit?

Reply:
Yes, good idea. I've push this to master and will rebase.